res/controllers/Behringer-CMDStudio4a-scripts.js

```diff
@@ -0,0 +1,257 @@
+// * Mixxx mapping script file for the Behringer CMD Studio 4a.
+// * Author: Craig Easton
+// * Version 1.2 (Jan 2016)
+// * Forum: http://www.mixxx.org/forums/viewtopic.php?f=7&amp;t=7868
+// * Wiki: http://www.mixxx.org/wiki/doku.php/behringer_cmd_studio_4a
+// ****************************************************************************
+
+////////////////////////////////////////////////////////////////////////
+// JSHint configuration                                               //
+////////////////////////////////////////////////////////////////////////
+/* global engine                                                      */
+/* global script                                                      */
+/* global print                                                       */
+/* global midi                                                        */
+//////////////////////////////////////////////////////////////////////// 
+
+// Master function definition.
+function BehringerCMDStudio4a() {}
+
+
+// ***************************** Global Vars **********************************
+
+// Shift/mode state variables.
+BehringerCMDStudio4a.delState = {0:false,1:false,2:false,3:false};
+BehringerCMDStudio4a.scratchState = {0:false,1:false,2:false,3:false};
+
+// Button push/release state variables.
+BehringerCMDStudio4a.pitchDec = {0:false,1:false,2:false,3:false};
+BehringerCMDStudio4a.pitchInc = {0:false,1:false,2:false,3:false};
+
+
+// ************************ Initialisation stuff. *****************************
+
+BehringerCMDStudio4a.vuMeterUpdate = function (value, group, control){
+    value=(value*15)+48;
+    switch(control) {
+    case "VuMeterL":
+        midi.sendShortMsg(0xB0, 0x7E, value);
+        break;
+    case "VuMeterR":
+        midi.sendShortMsg(0xB0, 0x7F, value);
+        break;
+    }
+}
+
+BehringerCMDStudio4a.initLEDs = function () {
+    // Initialise any LEDs that are direcctly controlled by this script.
+    // Del buttons (one for each virtual deck).
+    midi.sendShortMsg(0x90, 0x2A, 0x00);
+    midi.sendShortMsg(0x91, 0x4A, 0x00);
+    midi.sendShortMsg(0x92, 0x2A, 0x00);
+    midi.sendShortMsg(0x93, 0x4A, 0x00);
+    // Scratch buttons (one for each virtual deck).
+    midi.sendShortMsg(0x90, 0x16, 0x00);
+    midi.sendShortMsg(0x91, 0x36, 0x00);
+    midi.sendShortMsg(0x92, 0x16, 0x00);
+    midi.sendShortMsg(0x93, 0x36, 0x00);    
+}
+
+BehringerCMDStudio4a.init = function () {
+    // Initialise anything that might not be in the correct state.
+    BehringerCMDStudio4a.initLEDs();
+    // Connect the VUMeters
+    engine.connectControl("[Master]","VuMeterL","BehringerCMDStudio4a.vuMeterUpdate");
+    engine.connectControl("[Master]","VuMeterR","BehringerCMDStudio4a.vuMeterUpdate");
+}
+ 
+BehringerCMDStudio4a.shutdown = function() {
+    // Leave the deck in a properly initialised state.
+    BehringerCMDStudio4a.initLEDs();
+
+    // Disconnect the VUMeters.
+    // Maybe not! This doesn't seem to work here!

```

Comment:
What do you mean it doesn't work? What are you trying to accomplish? Generally, the shutdown function should turn off all LEDs.